### PR TITLE
[WIP]: handle duplicated blocks from entities with aliases and entity management

### LINKED ISSUE #22639: Adding duplicate Template Part removes both from page, can trigger clearing inner blocks in dirty state.
**Describe the bug**
This seems to be a recent regression.  Adding the same template part to a page twice will cause both to be removed from the page.  Navigating through the page content (up and down arrows) may then also trigger the same bug noted in #22638.

**To reproduce**
1. Go to the editor (page editor shown below) with full site editing experiment enabled.
2. Add a template part.
3. Add the same template part again.
4. See that both template parts are removed from the page.  Scroll (up and down arrows) through page content to see some 'funky' behavior and additionally trigger dirty state of the template part to have cleared all of its content (second part likely related to #22638) .

**Expected behavior**
Expected both template parts to remain (previous behavior).  Or if we are iterating to prevent that, at least 1 should remain and a notification should occur.

**Screenshots**
duplicate-template-part

Since this is a recent regression, perhaps @epiqueras, @youknowriad, @noahtallen, @vindl, @ockham, or others may have an idea of where this could originate from?

### PR DESCRIPTION
**Pull request by noahtallen**
## Description
This resolves #22639 by implementing the approach I described here: https://github.com/WordPress/gutenberg/pull/24180#issuecomment-729389698.

TLDR:
1. If a second inner block controller is inserted using the same entity ID as another, only the first inner block controller syncs changes in/out of the editor.
2. Internally, block edits are synced between the first and second inner block controllers with a new concept called "block aliases."

This way, edits are up to date inside the block tree, and we only have incoming/outgoing changes happen once. No need to handle multiple syncs or any of that nonsense!

## Tasks:
- [x] Implement an entity "management" tracker. This lets us know which template part block is supposed to be in control of the sync process.
- [x] Implement a "clone/alias/synchronize" block tree function, which lets us alias the block tree of the one template part under the other template part.
- [ ] Add state to keep track of block aliases.
- [x] When a template part block is not in charge of syncing, make sure it does not sync.
- [x] When a template part block is in charge of syncing, make sure it does sync.
- [x] When a template part block is not in charge of syncing, make sure it calls the function to alias its block tree.
- [ ] When a block is **added** under any template part, make sure its changes are synced to all of the alias blocks.
- [ ] When a block is **edited** under any template part, make sure its changes are synced to all of the alias blocks.
- [ ] When a block is **deleted** under any template part, make sure its changes are synced to all of the alias blocks.
- [ ] When a block is **deleted** under any template part, make sure its changes are synced to all of the alias blocks.
- [ ] Make sure that when a template part blocks is removed, the duplicate block becomes the primary manager.
- [ ] Performance testing
- [ ] Add unit tests
- [ ] Add documentation

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-editor/src/store/reducer.js [hunk 1]
```diff
@@ -716,6 +720,132 @@ const withSaveReusableBlock = ( reducer ) => ( state, action ) => {
...
+	'SHOW_INSERTION_POINT',
+	'SET_INSERTION_POINT',
+	'SELECT_BLOCK',
+	'START_DRAGGING_BLOCKS',
+];
+function getAliasedActions( state, action ) {
+	if ( actionsToIgnore.includes( action.type ) ) {
+		return;
+	}
+	if ( action.clientId ) {
```
Comment (noahtallen) [inline packages/block-editor/src/store/reducer.js:785]: This one handles things like updating block attributes: the action might only affect one client ID, so just apply it to the other aliased client IDs too. This however probably doesn't work for any action that might have clientId but also a root client ID 🤔 Thankfully most of those use `clientIds`.

**Comment by youknowriad** [comment]
superseded by  #27885 Thanks for you work on these issues @noahtallen
